### PR TITLE
Centos7 rpm installation guide

### PR DESCRIPTION
**Pull request by L-MA**
Created a installation guide based off the rpm packages from [PR #64 in the net-modules repo](https://github.com/mesosphere/net-modules/pull/64).

### CODE REVIEW

File: docs/DeployingOnCentOS7.md [hunk 1]
```diff
@@ -0,0 +1,291 @@
+# Deploying a Mesos Cluster with Calico on CentOS7
+
+In these instructions, we will install Mesos and Calico on CentOS/Red Hat Enterprise Linux 7 from RPM packages. These packages can be built from the Mesosphere [net-modules repo](https://github.com/mesosphere/net-modules) and will install the following processes:
```
Comment (djosborne) [inline docs/DeployingOnCentOS7.md]: ...on **baremetal** CentOS/Red Hat Enterprise Linux 7 from RPM packages...

File: docs/DeployingOnCentOS7.md [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+
+ * Mesos Master
+ * Mesos Agent
```
Comment (djosborne) [inline docs/DeployingOnCentOS7.md]: Also worth mentioning:
- net-modules

File: docs/DeployingOnCentOS7.md [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+## Build the packages
+
+The Mesos RPM packages can be built from the Mesosphere [net-modules repo](https://github.com/mesosphere/net-modules).
+
+    $ git clone git@github.com:mesosphere/net-modules.git
+    $ cd net-modules
+    $ make builder-rpm
+    $ ls packages/rpms/RPMS/x86_64
+
+The files you see in the `packages/rpms/RPMS/x86_64` directory are all the packages you need to install Mesos. Transfer these files to each host in your cluster.
```
Comment (djosborne) [inline docs/DeployingOnCentOS7.md]: ...you need to install Mesos, while additionally including the compiled net-modules library. Transfer these...

File: docs/DeployingOnCentOS7.md [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+
+    $ sudo yum -y install epel-release
+
+Now your server should be ready to install the Mesos packages.
+
+    $ sudo yum -y install *.rpm
+
+# Run the Backend Services
+
+You will need to run one instance of both, etcd and ZooKeeper. You can run etcd and ZooKeeper on any host you choose, but we recommend running them on your master.
```
Comment (djosborne) [inline docs/DeployingOnCentOS7.md]: We don't actually  *recommend* running them on master. Change paragraph this to indicate that most mesos clusters will run zookeeper (and now etcd since its a newly introduced calico dependency) in a cluster external to their core mesos cluster. In the name of simplicity, we'll be launching one instance of each as docker container's on our master.

Basically, get the message across that you'll never do this in production. We'll leave it up to you for how you want to do this step, but we'll show you the easiest simplest proof of concept way to do it.

File: docs/DeployingOnCentOS7.md [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+          ]
+        }
+      ]
+    }
+    EOF
+    $ sudo mv modules.json /calico/
+
+## Run Calico Node
+
+Download `calicoctl`, Calico's command line tool. We will use `calicoctl` to spin up calico-node in a docker container on each slave.
```
Comment (djosborne) [inline docs/DeployingOnCentOS7.md]: Most people aren't going to know what "spinning up a calico-node in a docker container" means. Reword this to give a glimpse of background for calico-node (Core calico-processes packaged up in a docker container).

File: docs/DeployingOnCentOS7.md [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+### Set Mesos' Master Environment Variables
+
+Append the following lines to `/etc/default/mesos-master` on each of your master hosts. 
+
+The value of `MESOS_QUORUM` may change depending on how many master hosts you have in your cluster. Mesos recommends that the quorum count is at least 1/2 the number of master hosts running. 
+
+Also note, you will need to fill in the IP address of the host that is running ZooKeeper.
+
+    MESOS_QUORUM=1
+    MESOS_WORK_DIR=/var/lib/mesos
```
Comment (djosborne) [inline docs/DeployingOnCentOS7.md]: `MESOS_WORK_DIR` and `MESOS_LOG_DIR` seem like they are vital to any mesos deployment, and thus should be already provided in the RPMs. We should investigate if they're necessary, what the defaults are, and if we should put them in the RPMs, leave them where they are, or remove them altogether. The other two are specific to our demo, so I'm fine with keeping them.

File: docs/DeployingOnCentOS7.md [hunk 7]
```diff
@@ -0,0 +1,291 @@
...
+## Mesos Slaves
+
+### Set Mesos' Slaves Environment Variables
+
+Append the following lines to `/etc/default/mesos-slave` on each of your slave hosts. 
+
+Note, you will need to fill in the IP address of the host that is running ZooKeeper and the IP address of the host running etcd.
+
+    MESOS_MASTER=zk://<IP of host with zookeeper running>:2181/mesos
+    MESOS_EXECUTOR_REGISTRATION_TIMEOUT=5mins
```
Comment (djosborne) [inline docs/DeployingOnCentOS7.md]: Same with these. I know we copied them from the docker-compose file, but we should weed out whichever ones aren't necessary.

File: docs/DeployingOnCentOS7.md (hunk 6 above)
Comment (djosborne) [inline docs/DeployingOnCentOS7.md]: I'm pretty sure the MESOS_LOG_DIR was changed to be the directory which was volume-mounted by docker-compose. Basically, we can let it be default since we're not volume mounting the  logs directory in this guide.

File: docs/DeployingOnCentOS7.md [hunk 8]
```diff
@@ -0,0 +1,291 @@
...
+
+## Run Calico Node
+
+Download `calicoctl`, Calico's command line tool. We will use `calicoctl` to spin up calico-node in a docker container on each slave. 
+
+When doing so, we must point `calicoctl` to our running instance of etcd. Go back and retrieve the IP address of the host where etcd is running. Use that address in the `ECTD_AUTHORITY` variable passed to `calicoctl`.
+
+    $ wget https://github.com/projectcalico/calico-docker/releases/download/v0.9.0/calicoctl
+    $ chmod +x calicoctl
+    $ sudo ETCD_AUTHORITY=<IP of host with etcd>:4001 ./calicoctl node
```
Comment (djosborne) [inline docs/DeployingOnCentOS7.md]: We'll want to provide a systemd service file for launching calico-node instead of a bash command

File: docs/DeployingOnCentOS7.md [hunk 9]
```diff
@@ -0,0 +1,291 @@
...
+
+If you followed the guide correctly so far, you should have the Mesos rpm packages on each of your master and slave hosts.
+
+Before installing these packages, you must have the Extra Packages for Enterprise Linux (or EPEL) packages installed.
+
+    $ sudo yum -y install epel-release
+
+Now your server should be ready to install the Mesos packages.
+
+    $ sudo yum -y install *.rpm
```
Comment (djosborne) [inline docs/DeployingOnCentOS7.md]: We should split this up so master only installs the master packages, and the slave only installs the slave packages.

Comment (djosborne) [inline docs/DeployingOnCentOS7.md]: Ah, I see that that's not the way the packages are divided. Looks like we have to install all packages on both.

File: docs/DeployingOnCentOS7.md [hunk 10]
```diff
@@ -0,0 +1,291 @@
...
+Calico will need to be running on each of your slaves in order to perform Calico networking. 
+
+To obtain the Calico files, you will need `wget` installed. If you haven't already done so, download the tool with `yum -y install wget`.
+
+## Download the Calico Isolator
+
+    $ wget https://github.com/projectcalico/calico-mesos/releases/download/v0.1.1/calico_mesos
+    $ chmod +x calico-mesos
+    $ sudo mkdir /calico
+    $ sudo mv calico_mesos /calico/calico_isolator
```
Comment (djosborne) [inline docs/DeployingOnCentOS7.md]: It used to be called `calico_isolator` in 0.24 but we've renamed it to `calico_mesos` for 0.25. So update this line as well as the values in the json file to use calico_mesos.

File: docs/DeployingOnCentOS7.md [hunk 11]
```diff
@@ -0,0 +1,293 @@
...
+
+    $ sudo mkdir /framework
+    $ cd /framework
+    $ sudo wget https://raw.githubusercontent.com/mesosphere/net-modules/integration/0.25/framework/calico_executor.py
+    $ sudo wget https://raw.githubusercontent.com/mesosphere/net-modules/integration/0.25/framework/calico_framework.py
+    $ sudo wget https://raw.githubusercontent.com/mesosphere/net-modules/integration/0.25/framework/calico_utils.py
+    $ sudo wget https://raw.githubusercontent.com/mesosphere/net-modules/integration/0.25/framework/constants.py
+    $ sudo wget https://raw.githubusercontent.com/mesosphere/net-modules/integration/0.25/framework/tasks.py
+
+Check to see if your system supports the `ifconfig` command and the `nc` command. If not, you will need to download these packages on each host.
```
Comment (djosborne) [inline docs/DeployingOnCentOS7.md]: We don't use `ifconfig` anymore with your change to use `ip addr`, right?

File: docs/DeployingOnCentOS7.md [hunk 12]
```diff
@@ -0,0 +1,293 @@
...
+
+    $ sudo yum install net-tools
+    $ sudo yum install nc
+
+Next, we need to hardcode the master IP in `calico_framework.py` at `L548`, so it reads `master_ip = "<IP of master host>:5050"`. This change only needs to happen on the master.
```
Comment (djosborne) [inline docs/DeployingOnCentOS7.md]: This isn't true either, right?

File: docs/DeployingOnCentOS7.md [hunk 13]
```diff
@@ -0,0 +1,274 @@
+# Deploying a Mesos Cluster with Calico on CentOS7
+
+In these instructions, we will install Mesos and Calico Networking on baremetal CentOS/Red Hat Enterprise Linux 7 from rpm packages. These packages can be built from the Mesosphere [net-modules repo](https://github.com/mesosphere/net-modules) and will install the following processes:
```
Comment (djosborne) [inline docs/DeployingOnCentOS7.md]: Remove "Red Hat Enterprise Linux" from here. The two are actually different, and this install is for CentOS7.

File: docs/DeployingOnCentOS7.md [hunk 14]
```diff
@@ -0,0 +1,274 @@
...
+
+ * Mesos Master
```
Comment (djosborne) [inline docs/DeployingOnCentOS7.md]: Lets ditch this bulleted list. Its hard to cover all the stuff this guide will install on your system and what type it is in just a few sentences. Lets try:

In these instructions, we will set up a Mesos Cluster with Calico Networking on bare-metal CentOS7 using RPMs built by [net-modules](https://github.com/mesosphere/net-modules). These RPMs conveniently package Mesos with compiled net-modules code.

File: docs/DeployingOnCentOS7.md [hunk 15]
```diff
@@ -0,0 +1,274 @@
...
+
+## Preparation
+
+For the installation of Mesos + Calico, you will need Docker 1.7+. You must also verify your host has a unique fully qualified domain name. Lastly, we recommend disabling the firewall on the Mesos communication ports.
+
+The instruction below will walk you through the steps.
+
+### Prepare External Services
+
+You will need to run one instance of both, etcd and ZooKeeper. Most Mesos cluster will run the backend services in a cluster external to their core Mesos cluster. For simplicity's sake, we will be launching ZooKeeper and etcd on the master host.
```
Comment (djosborne) [inline docs/DeployingOnCentOS7.md]: Zookeeper and etcd serve as the backend datastores for Mesos and Calico, respectively. Most Mesos clusters will run these services in a seperate cluster, seperate from the Masters and Agents which read from them. 

As a simple proof-of-concept, we'll cover running one instance of each as Docker containers on our Mesos-Master. However, you are welcome to run them wherever you choose.

**Comment by L-MA** [comment]
Closing since this was addressed with this PR: https://github.com/projectcalico/calico-docker/pull/615
